include/envoy/registry/registry.h

```diff
@@ -231,6 +278,26 @@ template <class Base> class FactoryRegistry : public Logger::Loggable<Logger::Id
     }
   }
 
+  /**
+   * @return true if the named factory was disabled.
+   */
+  static bool isFactoryDisabled(absl::string_view name) {
+    auto it = factories().find(name);
+    return it == factories().end() || it->second == nullptr;
```

Comment:
Is it semantically correct to say that a missing extension is disabled?

Reply:
I think this method should not be called if a factory was not registered. I changed it so it ASSERTs on failed precondition.